### ISSUE TITLE
Exporting all pages/data, not only first page or filtered results

### ISSUE THREAD

**Issue by punchi** [hhurz/tableExport.jquery.plugin#234]
I have a table with 20 pages and each page contains 150 records. By using this plugin I'm trying to export the complete table data, but it is exporting only first page or filtered results (but filtered is fine!). To export all the 3.000 records I need to set "lengthMenu" to -1 (or all) but takes forever to load and some PC's can not handle that. It is possible to export all the pages? it is possible to add this functionality? right now only exports "what it shown" thanks!! =)

**Comment by luyaozhang** [comment]
Try like this, but can not export too large table ,  rows index lager than 10000.
Because 'togglePagination' may block page.

function exportTable() {
$('#table').bootstrapTable('togglePagination');
$('#table').tableExport({
        type: 'excel',
        fileName:  'XXX',
        escape: 'false'
    });
$('#table').bootstrapTable('togglePagination');
}

**Comment by hhurz** [comment]
Thats a often asked question. Please see the comments in these issues: #122, #85, #54, and #26

**Comment by punchi** [comment]
I'm using datatables. The problem is not solved with #85 and #122
As for #54, any news related this?
>  Anyway, I'm thinking about to provide such a callback mechanism in the near future...

Regaring #26 is the only viable way right now.
By the while, with this code with datatables
```
"buttons": [
    {
        extend: 'excelHtml5',
        exportOptions: {
            columns: [ 1,2,3,4,6,7,8 ],
            format: {
                body: function ( data, row, column, node ) {
                    // Strip $ from salary column to make it numeric
                    return column === 6 || column === 4 ?
                        data.replace( /[,.]/g, '' ):data;
                }
            }
        }
    }
]
```
I can export to XLS all rows, but still looking how to integrate this call to the tableExport menu.
Thanks @luyaozhang and @hhurz
